### PR TITLE
Category List in IFrame

### PR DESCRIPTION
**Pull request by wazolab**

### CODE REVIEW

File: components/PoisList/PoisTable.vue [hunk 1]
```diff
@@ -171,11 +177,11 @@ function getContext(key: string) {
...
           <h1 class="d-flex align-center print:tw-pb-4" :style="{ marginRight: device.smallScreen ? 'unset' : 'auto' }">
             <TeritorioIconBadge
-              :color-fill="category.category.color_fill"
-              :picto="category.category.icon"
+              :color-fill="category?.category.color_fill || ''"
+              :picto="category?.category.icon || ''"
               size="xl"
             />
-            {{ category.category.name.fr }}
+            {{ category?.category.name.fr }}
```
Comment (frodrigo) [inline components/PoisList/PoisTable.vue:184]: Make this part conditioned by a `v-if="category"` ?

File: components/PoisList/PoisTable.vue [hunk 2]
```diff
@@ -35,24 +34,26 @@ const search = ref('')
...
-    props.category.id,
-    { geometry_as: 'point', short_description: true },
-  ))
-
-  if (error.value || !data.value)
-    throw createError({ statusCode: 404, statusMessage: 'POIs not found.' })
-
-  pois.value = data.value
-  loadingState.value = pending.value
+if (props.category) {
```
Comment (frodrigo) [inline components/PoisList/PoisTable.vue:38]: Check for `!== undefined` and no  `props.category!.id` (`!`) latter?

File: components/PoisList/PoisTable.vue (hunk 1 above)
Comment (wazolab) [reply components/PoisList/PoisTable.vue:184]: I didn't in ordre to keep 'empty' UI and make the use of default props instead.
I give a 'more' consistent UI to user's eye and reduce Layout Shift as well.

It was also a thinking about whether to have en ampty page or page with empty state UI.

File: components/PoisList/PoisTable.vue [hunk 3]
```diff
@@ -35,24 +34,26 @@ const search = ref('')
 // Fetch POIs by Cache or API
 const pois = ref()
```
Comment (frodrigo) [inline components/PoisList/PoisTable.vue:35]: Missing typing, from previous PR.

File: components/PoisList/PoisTable.vue (hunk 2 above)
Comment (wazolab) [reply components/PoisList/PoisTable.vue:38]: In here I think we are facing Typings limitations, even if I add `!== undefined` TS will shout that `props.category` is potentially `undefined` and we are within a `computed` or arrow function (which seems to break type inferance ...)

File: components/PoisList/PoisTable.vue (hunk 3 above)
Comment (wazolab) [reply components/PoisList/PoisTable.vue:35]: mmh yep `ApiPois`

File: components/PoisList/PoisTable.vue (hunk 2 above)
Comment (frodrigo) [reply components/PoisList/PoisTable.vue:38]: providing empty color and icon is unexpected for TeritorioIconBadge, it shouldn't event accept that.

Comment (frodrigo) [reply components/PoisList/PoisTable.vue:38]: Maybe use a default icon and color? eg `services-information`

**Comment by frodrigo** [comment]
- Requering a category give a 404
- The icon is missing (but...)

Capture d’écran 2024-03-19 à 18 44 54

If we want to keep a constant display we have to keep the export icon (as disabled). But I find this empty table ugly.

I f we support empty here let's be homogeneous with `/category/{id}` and be the same here.

- Could we make the table fill the page (height) ?
- Like on the `/embedded` make the selector horizontally fill the page.
- Selector with no (top) padding could look ugly on page embedding the iframe

cc @sbterito

**Comment by frodrigo** [comment]
> Selector with no (top) padding could look ugly on page embedding the iframe
- [ ] Need same padding on right/left, unless we still have the same issue.

- [ ] The icon default for empty is missing

**Comment by wazolab** [comment]
**The icon default for empty is missing**
It's weird, it work on my side ... can you provide more details ? console error maybe ?

**Comment by frodrigo** [comment]
> The icon default for empty is missing

It does not works, because I test with only instance not using the teritorio font. And thath good! We cannot hard code ref to teritorio font. Here we have to use fontawesome.

**Comment by wazolab** [comment]
> > The icon default for empty is missing
> 
> It does not works, because I test with only instance not using the teritorio font. And thath good! We cannot hard code ref to teritorio font. Here we have to use fontawesome.

It should be ok now !

**Pull request merged**